ui/components/app/detected-token/detected-token.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+
+import {
+  importTokens,
+  ignoreTokens,
+  setNewTokensImported,
+} from '../../../store/actions';
+import { getDetectedTokensInCurrentNetwork } from '../../../selectors';
+
+import DetectedTokenSelectionPopover from './detected-token-selection-popover/detected-token-selection-popover';
+import DetectedTokenIgnoredPopover from './detected-token-ignored-popover/detected-token-ignored-popover';
+
+const DetectedToken = ({ setShowDetectedTokens }) => {
+  const dispatch = useDispatch();
+
+  const detectedTokens = useSelector(getDetectedTokensInCurrentNetwork);
+
+  const [selectedTokens, setSelectedTokens] = useState(() =>
+    detectedTokens.reduce((tokenObj, token) => {
+      tokenObj[token.address] = token;
+      return tokenObj;
+    }, {}),
+  );
+  const [
+    showDetectedTokenIgnoredPopover,
+    setShowDetectedTokenIgnoredPopover,
+  ] = useState(false);
+
+  const handleClearTokensSelection = async () => {
+    const selectedTokensList = Object.values(selectedTokens);
+    const unSelectedTokensList = detectedTokens.filter(
+      (token) => !Object.keys(selectedTokens).includes(token.address),
+    );
+
+    if (selectedTokensList.length < detectedTokens.length) {
+      await dispatch(ignoreTokens(unSelectedTokensList));
+      await dispatch(importTokens(selectedTokensList));
+      const tokenSymbols = selectedTokensList.map(({ symbol }) => symbol);
+      dispatch(setNewTokensImported(tokenSymbols.join(', ')));
+    } else {
+      await dispatch(ignoreTokens(unSelectedTokensList));
+    }
+    setShowDetectedTokens(false);
+  };
+
+  const handleTokenSelection = (token) => {
+    const newSelectedTokens = { ...selectedTokens };
+
+    if (selectedTokens[token.address]) {
+      delete newSelectedTokens[token.address];
+    } else {
+      newSelectedTokens[token.address] = token;
+    }
+    setSelectedTokens(newSelectedTokens);
```

Comment:
Getting better but  I still think we could make it better by 

1. `selectedTokensList` -> `tokensList`
2. `[token.address]: token` -> `[token.address]: true/false`
3. delete unSelectedTokensList because a false value indicates it is not selected

then inside handleTokenSelection you can just flip the boolean value instead of adding/deleting entries. 

If you *need* the token itself too, you can switch this to: 

```
[token.address]: { token. selected: true/false}
```